socorro/processor/processor_2015.py

```diff
@@ -158,19 +176,46 @@ def __init__(self, config, quit_check_callback=None):
         else:
             self.quit_check = lambda: False
 
-        # here we instantiate the rule sets and their rules.
-        self.rule_system = OrderedDotDict()
-        for a_rule_set_name in config.rule_sets.names:
-            self.config.logger.debug(
-                'setting up rule set: %s',
-                a_rule_set_name
-            )
-            self.rule_system[a_rule_set_name] = (
-                config[a_rule_set_name].rule_system_class(
-                    config[a_rule_set_name],
-                    self.quit_check
-                )
-            )
+        rule_set = rules or [
```

Comment:
I'd prefer if this default set of rules was at the top of the module with a name like `DEFAULT_RULES` because it feels like a "constant", but buried in the code.

Reply:
moved it out to a module level constant in the latest patch, and then made it the default arg. This isn't the same pattern as the one used in the `SignatureGenerator` class. Not sure if one or the other is preferable.

Reply:
The SignatureGenerator is different in that it has no configman anything in it. Is that what you mean by "it's not the same pattern"? If not, can you explain a bit more?

Reply:
SignatureGenerator uses something like:

```
def __init__(self, rules=None ...):
    self.rules = rules or list(DEFAULT_RULES)
```

where instead I used something like:

```
def __init__(self, rules=DEFAULT_RULES ...):
    self.rules = rules
```

Reply:
Ohhh! Ok. So, I think the difference between those two is a style thing. In the SignatureGenerator case (which I wrote, so I'm using my style), I did what I did because of an annoying thing about Python where argument defaults get created at module load time and persist, so it's good to avoid using mutable values as defaults. Additionally, if Sphinx autodoc runs over code like this:

```
DEFAULT_RULES = [some long list of things]

def __init__(self, rules=DEFAULT_RULES):
    ...
```

it generates a (in my opinion) unreadable signature. For example:

https://bleach.readthedocs.io/en/latest/clean.html#bleach.clean

I think what you're doing and what I did in SignatureGenerator are behaviorally equivalent. I'm game for whichever appeals to you.

Reply:
cool. In the latest patch I've updated it to match SignatureGenerator